modules/utxo-lib/src/bitgo/zcash/ZcashPsbt.ts

```diff
@@ -0,0 +1,73 @@
+import { PsbtOpts, UtxoPsbt } from '../UtxoPsbt';
+import {
+  getDefaultConsensusBranchIdForVersion,
+  getDefaultVersionGroupIdForVersion,
+  ZcashTransaction,
+} from './ZcashTransaction';
+import { Network, PsbtTransaction } from '../../';
+import { Psbt as PsbtBase } from 'bip174';
+import * as types from 'bitcoinjs-lib/src/types';
+
+const typeforce = require('typeforce');
+
+export class ZcashPsbt extends UtxoPsbt<ZcashTransaction<bigint>> {
+  protected static transactionFromBuffer(buffer: Buffer, network: Network): ZcashTransaction<bigint> {
+    return ZcashTransaction.fromBuffer<bigint>(buffer, false, 'bigint', network);
+  }
+
+  static createPsbt(opts: PsbtOpts, data: PsbtBase): ZcashPsbt {
+    return new ZcashPsbt(
+      opts,
+      data || new PsbtBase(new PsbtTransaction({ tx: new ZcashTransaction<bigint>(opts.network) }))
+    );
+  }
+
+  setVersion(version: number, overwinter = true): this {
+    typeforce(types.UInt32, version);
+    this.tx.overwintered = overwinter ? 1 : 0;
+    this.tx.version = version;
+    return this;
+  }
+
+  setDefaultsForVersion(network: Network, version: number): void {
+    switch (version) {
+      case 4:
+      case ZcashTransaction.VERSION4_BRANCH_CANOPY:
+      case ZcashTransaction.VERSION4_BRANCH_NU5:
+        this.setVersion(4);
+        break;
+      case 5:
+      case ZcashTransaction.VERSION5_BRANCH_NU5:
+        this.setVersion(5);
+        break;
+      default:
+        throw new Error(`invalid version ${version}`);
+    }
+
+    this.tx.versionGroupId = getDefaultVersionGroupIdForVersion(version);
+    this.tx.consensusBranchId = getDefaultConsensusBranchIdForVersion(network, version);
+  }
+
+  private setPropertyCheckSignatures(propName: keyof ZcashTransaction<bigint>, value: unknown) {
+    if (this.tx[propName] === value) {
+      return;
+    }
+    this.checkForSignatures(propName);
+    this.tx[propName] = value as any;
```

Comment:
Do you know why we declare `value` as an `unknown` and then just cast it to any without checking its type at all? It seems to circumvent the purpose of `unknown` and we might as well just declare `value: any` up above.

I see we were already doing this in ZcashTransactionBuilder but it seems odd to me.

Reply:
I don't know why. Yeah, we just copied this over. Might kill it later as it's less useful in PSBT than TxBuilder.

Reply:
yeah, could probably type this better as `typeof keyof ZcashTransaction<bigint>`, but I think killing this is the right thing to do long term anyway